# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.h
# Commit: b9d32ef0ad83
# Full Hash: b9d32ef0ad835c014d1ced7577ed3d4c47a00031
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Bulk memory operations. r=yury
#   
#   Implement all bulk memory operations on 64-bit memories.  Only
#   out-of-line paths are supported for memory.copy and memory.fill, for
#   now.
# ==============================================================================

diff -r 359e414a1d7f -r b9d32ef0ad83 js/src/wasm/WasmBuiltins.h
--- a/js/src/wasm/WasmBuiltins.h	Mon Oct 18 07:31:54 2021 +0000
+++ b/js/src/wasm/WasmBuiltins.h	Mon Oct 18 07:31:55 2021 +0000
@@ -87,17 +87,24 @@
   Uint64ToDouble,
   Int64ToFloat32,
   Int64ToDouble,
-  MemoryGrow,
-  MemorySize,
+  MemoryGrowM32,
+  MemoryGrowM64,
+  MemorySizeM32,
+  MemorySizeM64,
   WaitI32,
   WaitI64,
   Wake,
-  MemCopy32,
-  MemCopyShared32,
+  MemCopyM32,
+  MemCopySharedM32,
+  MemCopyM64,
+  MemCopySharedM64,
   DataDrop,
-  MemFill32,
-  MemFillShared32,
-  MemInit32,
+  MemFillM32,
+  MemFillSharedM32,
+  MemFillM64,
+  MemFillSharedM64,
+  MemInitM32,
+  MemInitM64,
   TableCopy,
   ElemDrop,
   TableFill,
@@ -203,17 +210,24 @@
 extern const SymbolicAddressSignature SASigLogD;
 extern const SymbolicAddressSignature SASigPowD;
 extern const SymbolicAddressSignature SASigATan2D;
-extern const SymbolicAddressSignature SASigMemoryGrow;
-extern const SymbolicAddressSignature SASigMemorySize;
+extern const SymbolicAddressSignature SASigMemoryGrowM32;
+extern const SymbolicAddressSignature SASigMemoryGrowM64;
+extern const SymbolicAddressSignature SASigMemorySizeM32;
+extern const SymbolicAddressSignature SASigMemorySizeM64;
 extern const SymbolicAddressSignature SASigWaitI32;
 extern const SymbolicAddressSignature SASigWaitI64;
 extern const SymbolicAddressSignature SASigWake;
-extern const SymbolicAddressSignature SASigMemCopy32;
-extern const SymbolicAddressSignature SASigMemCopyShared32;
+extern const SymbolicAddressSignature SASigMemCopyM32;
+extern const SymbolicAddressSignature SASigMemCopySharedM32;
+extern const SymbolicAddressSignature SASigMemCopyM64;
+extern const SymbolicAddressSignature SASigMemCopySharedM64;
 extern const SymbolicAddressSignature SASigDataDrop;
-extern const SymbolicAddressSignature SASigMemFill32;
-extern const SymbolicAddressSignature SASigMemFillShared32;
-extern const SymbolicAddressSignature SASigMemInit32;
+extern const SymbolicAddressSignature SASigMemFillM32;
+extern const SymbolicAddressSignature SASigMemFillSharedM32;
+extern const SymbolicAddressSignature SASigMemFillM64;
+extern const SymbolicAddressSignature SASigMemFillSharedM64;
+extern const SymbolicAddressSignature SASigMemInitM32;
+extern const SymbolicAddressSignature SASigMemInitM64;
 extern const SymbolicAddressSignature SASigTableCopy;
 extern const SymbolicAddressSignature SASigElemDrop;
 extern const SymbolicAddressSignature SASigTableFill;